compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1402,16 +1405,17 @@ impl NodeInterner {
                 });
 
                 if trait_id.is_none() && matches!(self_type, Type::DataType(..)) {
-                    if let Some(existing) = self.lookup_direct_method(self_type, &method_name, true)
+                    let check_self_param = false;
+                    if let Some(existing) =
+                        self.lookup_direct_method(self_type, &method_name, check_self_param)
```

Comment:
Reviewing atm, but flagging up that this has flipped.

Reply:
Right, that's the fix :-)

Previously we errors on a duplicate method when the type of `self` was the same as the previous method. The problem is that when a method has no `self` it wouldn't error. So there's no need to check the type of `self`. However, we do need to check that the method belongs to the same type to error, and that's what the last commit does.